OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -134,6 +135,7 @@ public Armament(Actor self, ArmamentInfo info)
 				barrels.Add(new Barrel { Offset = WVec.Zero, Yaw = WAngle.Zero });
 
 			Barrels = barrels.ToArray();
+			attack = self.TraitsImplementing<AttackBase>().FirstOrDefault(a => a.Info.Armaments.Contains(Info.Name));
```

Comment:
I am not sure about this - AttackBase has AttackTargetCenter, some Projectiles have TargetCenterPosition, here Armament uses one of AttackBases. Can it happen that 1 actor has more Attacks with the same Armament? Then this can select incorrect one and use wrong AttackTargetCenter.